sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -3828,6 +3828,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_ARRAY_SORT_FAILS_ON_NULL_COMPARISON_RESULT =
+    buildConf("spark.sql.legacy.arraySortFailsOnNullComparisonResult")
```

Comment:
I feel the "fail" behavior is the legacy one from the new conf name. The previous name was good. Or we can follow the other conf style `spark.sql.legacy.allowNullComparisonResultInArraySort`.
Sorry about the previous trivial comment: https://github.com/apache/spark/pull/36812#discussion_r892996318

Reply:
No worries, thanks for the suggestions!
`spark.sql.legacy.allowNullComparisonResultInArraySort` sounds better to me.